app/schema/api/v4/models/questionnaire.rb

```diff
@@ -0,0 +1,65 @@
+class Api::V4::Models::Questionnaire
```

Comment:
This is the part that I don't understand - are we baking in the _screening_ form schema into this? How does that work with us updating the schema later? Will we update API versions if we make breaking changes? Additionally, this endpoint is the questionnaire endpoint, not the screening endpoint, right? Shouldn't it be more generic?

I imagined that this model would describe the general form DSL, and would allow screening and other forms to be exchanged - also allowing us to update the _screening_ form version, without having to update the _API_ version. Could you help me understand this?

Reply:
Hey Hari, addressing this 1 by 1.

> Additionally, this endpoint is the questionnaire endpoint, not the screening endpoint
Modified to `Models::Questionnaire::MonthlyScreeningReports`

> I imagined that this model would describe the general form DSL
As we add more forms, we'll update API-docs to contain a specimen layout for each DSL. We cannot generalise this using Swagger as random example strings won't make sense without the context it has currently.

> Will we update API versions if we make breaking changes?
We can go about this in 3 ways. Prabhanshu & I are still not decided on this. We'll loop Govind & you tomorrow (Thu, 15 Dec) if we don't have clarity. The 3 options:

1. We send all form-layout versions to the app. App chooses which one to render
2. We bump up api-version from v4 -> v5
3. We store minimum supported app-version on server-side

Reply:
> Will we update API versions if we make breaking changes?

@harimohanraj89 we've decided to go ahead with a questionnaire_api_version to solve this problem. Refer to Screening PRD header: [API Version changes](https://docs.google.com/document/d/1oRL3RyE6BO4kMhDLxjmWSHMwtKVnknxBGuYu9yyVwn4/edit#heading=h.fqghw6d73kj8) for more details

Reply:
Regarding Swagger, I understand. However, it seems a bit weird to put a specific screening schema into the Swagger spec for this generic API. I did a 10 second google for how to express a generic recursive schema in Swagger, and found this: https://stackoverflow.com/questions/36866035/how-to-refer-to-enclosing-type-definition-recursively-in-openapi-swagger

Could we use something like this so that our API spec and docs accurately represent the generic questionnaire schema, and we can enrich the API docs with an example?

Reply:
Rename: ✅ 

Questionnaire API version: Looks good. I've left some clarifying questions (for my own understanding) in the PRD itself. This looks like a good approach.